### DISCUSSION TITLE
Programatic FluentComboBox in C#?

### DISCUSSION THREAD

**Discussion by thoffmann-fms** [microsoft/fluentui-blazor#212] [Q&A] [answered]
Sorry, very newbie question.  How do I generate the FluentOption values at runtime?
e.g. Something like:
`                    <FluentCombobox>
                        @(GetRegions())
                    </FluentCombobox>
`
and 

`   
     private List<FluentOption> GetRegions()
    {
        FluentOption fo = new FluentOption();
        fo.Name = "test";
        fo.Value = "1";

        List<FluentOption> result = new List<FluentOption>();
        result.Add(fo);
        return result;
    }
`

**Comment by vnbaaij** [comment, marked as answer]
Hi,
You can do it like this:
```
<h2>From list of string items</h2>
<FluentCombobox Options=@stringOptions />

<h2>From list of int items</h2>
<FluentCombobox Options=@intOptions />

@code{
    List<Option<string>> stringOptions = new()
        {
            { new Option<string> { Key = "1", Value = "One" } },
            { new Option<string> { Key = "2", Value = "Two", Selected = true } },
            { new Option<string> { Key = "3", Value = "Three" } }

        };

    List<Option<int>> intOptions = new()
        {
            { new Option<int> { Key = 1, Value = 1, Disabled = true } },
            { new Option<int> { Key = 2, Value = 2 } },
            { new Option<int> { Key = 3, Value = 3 } }

        };
}
```

**Comment by thoffmann-fms** [comment]
Thanks, but what namespace is “Option” in?

e.g.
@using Microsoft.Fast.Components.FluentUI;
<snip>
List<Option<int>> GetRegions = new>();

does not compile.

Tim Hoffmann
Sr. Software Engineer
(231) 252-1202
FM:Systems

**Reply by vnbaaij** [reply]
You have new>() instead of new(). `Option` is in the same namespace as everything else.

**Reply by thoffmann-fms** [reply]
Given these using statements, Option is not defined. Using .NET 6.0.
```
@using Microsoft.Extensions.Configuration
@using Microsoft.AspNetCore.Hosting
@using Microsoft.Extensions.Hosting
@using Microsoft.Fast.Components.FluentUI
@using Microsoft.Fast.Components;
@using Microsoft.Data.SqlClient;
```



![image](https://user-images.githubusercontent.com/28546239/189921462-f657e0ca-6b0f-4d3a-8503-8bf033d16556.png)

**Reply by vnbaaij** [reply]
Judging from you screenshot, you are using the Teams Toolkit, right? Maybe there is a collision with it also defined somewhere else. Can you try with using the full name (Microsoft.Fast.Components.FluentUI.Option)

**Reply by thoffmann-fms** [reply]
Duh, the Teams template I'm playing with references V1 of the FluentUI NuGet package... Updated to 1.5.3 and Option is now defined...  Thanks for the assistance!
